Sources/NimiqClient/NimiqClient.swift

```diff
@@ -0,0 +1,1082 @@
+
+    /// Returns or overrides a constant value.
+    /// When no parameter is given, it returns the value of the constant. When giving a value as parameter,
+    /// it sets the constant to the given value. To reset the constant use `resetConstant()` instead.
+    /// - Parameter string: The class and name of the constant (format should be `Class.CONSTANT`).
+    /// - Parameter value: The new value of the constant.
+    /// - Returns: The value of the constant.
+    public func constant(_ constant: String, value: Int? = nil) throws -> Int? {
+        var params:[Any] = [constant]
+        if value != nil {
+            params.append(value!)
+        }
+        return try call(method: "constant", params: params)
+    }
+
+    /// Creates a new account and stores its private key in the client store.
+    /// - Returns: Information on the wallet that was created using the command.
+    public func createAccount() throws -> Wallet? {
+        return try call(method: "createAccount", params: [])
+    }
+
+    /// Creates and signs a transaction without sending it. The transaction can then be send via `sendRawTransaction()` without accidentally replaying it.
+    /// - Parameter transaction: The transaction object.
+    /// - Returns: Hex-encoded transaction.
+    public func createRawTransaction(_ transaction: OutgoingTransaction) throws -> String? {
+        let params:[String:Any?] = [
+            "from": transaction.from,
+            "fromType": transaction.fromType?.rawValue,
+            "to": transaction.to,
+            "toType": transaction.toType?.rawValue,
+            "value": transaction.value,
+            "fee": transaction.fee,
+            "data": transaction.data
+        ]
+        return try call(method: "createRawTransaction", params: [params])
+    }
+
+    /// Returns details for the account of given address.
+    /// - Parameter address: Address to get account details.
+    /// - Returns: Details about the account. Returns the default empty basic account for non-existing accounts.
+    public func getAccount(address: Address) throws -> Any? {
+        let result: RawAccount = try call(method: "getAccount", params: [address])!
+        return result.value
+    }
+
+    /// Returns the balance of the account of given address.
+    /// - Parameter address: Address to check for balance.
+    /// - Returns: The current balance at the specified address (in smalest unit).
+    public func getBalance(address: Address) throws -> Int? {
+        return try call(method: "getBalance", params: [address])
+    }
+
+    /// Returns information about a block by hash.
+    /// - Parameter hash: Hash of the block to gather information on.
+    /// - Parameter fullTransactions: If `true` it returns the full transaction objects, if `false` only the hashes of the transactions.
+    /// - Returns: A block object or `nil` when no block was found.
+    public func getBlockByHash(_ hash: Hash, fullTransactions: Bool = false) throws -> Block? {
```

Comment:
nitpick (non-blocking): this could be refactored to have "fluent design".

Another example of fluent design, this could be refactored to `public func block(forHash hash: Hash, withTransactions fullTransactions)`resulting in this usage: `block(forHash: blockHash, withTransactions: true)`

Reply:
I like the fluent design, that's for sure 👍

Reply:
Method names now adhere to Fluente Design, this has been refactored into `getBlock(forHash hash:, withTransactions:)`